packages/platform/setup-accounts.js

```diff
@@ -82,29 +92,27 @@ export default ({ mergeUserCartsOnLogin = true } = {}) => {
     check(email, Match.OneOf(String, null, undefined));
     const guestname = `${moniker.choose()}-${Random.hexString(5)}`;
     return {
-      email: email || `${guestname}@localhost`,
+      email: email || `${guestname}@unchained.local`,
       guest: true,
       profile: {},
     };
   }
 
-  Accounts.registerLoginHandler('guest', (options) => {
-    if (!options || !options.createGuest) {
-      return undefined;
-    }
-    const guestOptions = createGuestOptions(options.email);
-    return {
-      userId: Accounts.createUser(guestOptions),
-    };
-  });
+  accountsServer.services.guest = {
+    async authenticate(params) {
+      const guestOptions = createGuestOptions(params.email);
```

Comment:
If you throw here it is the equivalent of throwing in accountsServer.on('ValidateLogin' so I guess you could move the code in ValidateLogin to here and make it more clear what happens when it's a guest login